test/unit/BPool/BPool_ExitPool.t.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.25;
+
+import {BPoolBase} from './BPoolBase.sol';
+
+import {IERC20Errors} from '@openzeppelin/contracts/interfaces/draft-IERC6093.sol';
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+
+import {BNum} from 'contracts/BNum.sol';
+import {IBPool} from 'interfaces/IBPool.sol';
+
+contract BPoolExitPool is BPoolBase {
+  // Valid scenario
+  uint256 public constant SHARE_PROPORTION = 20;
+  uint256 public poolAmountIn = INIT_POOL_SUPPLY / SHARE_PROPORTION;
+  uint256 public token0Balance = 20e18;
+  uint256 public token1Balance = 50e18;
+  uint256[] minAmountsOut;
+
+  // when buring n pool shares, caller expects enough amount X of every token t
+  // should be sent to statisfy:
+  // Xt = n/BPT.totalSupply() * t.balanceOf(BPT)
+  uint256 public expectedToken0Out = token0Balance / SHARE_PROPORTION;
+  uint256 public expectedToken1Out = token1Balance / SHARE_PROPORTION;
+
+  function setUp() public virtual override {
+    super.setUp();
+    bPool.set__finalized(true);
+    // mint an initial amount of pool shares (expected to happen at _finalize)
+    bPool.call__mintPoolShare(INIT_POOL_SUPPLY);
+    address[] memory _tokens = new address[](2);
+    _tokens[0] = token;
+    _tokens[1] = secondToken;
```

Comment:
how do you feel about `token0` and `token1`?

Reply:
seems reasonable, I could even set up the arrays on the base contract, since that's essentially repeated on every test contract. (but not setting the token array in the pool, since that'd break joinpool tests for example)